test/SlowTests/Voron/Issues/RavenDB_16536.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using FastTests.Voron;
+using Sparrow;
+using Voron;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Voron.Issues
+{
+    public class RavenDB_16536 : StorageTest
+    {
+        public RavenDB_16536(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        protected override void Configure(StorageEnvironmentOptions options)
+        {
+            options.ManualFlushing = true;
+            options.ManualSyncing = true;
+            options.MaxNumberOfPagesInJournalBeforeFlush = 2; // low value to ensure it will flush
+        }
+
+        [Fact]
+        public void ShouldNotSayThatThereIsNothingToFlush()
+        {
+            for (int i = 0; i < 100; i++)
+            {
+                using (var tx = Env.WriteTransaction())
+                {
+                    var tree = tx.CreateTree("items");
+
+                    tree.Add("items/" + i, new byte[] { 1, 2, 3 });
+
+                    tx.Commit();
+                }
+            }
+
+            var onLogsAppliedMre = new ManualResetEventSlim(false);
+
+            Env.OnLogsApplied += () => onLogsAppliedMre.Set();
+
+            var ownFlusher = new GlobalFlushingBehavior();
+
+            ownFlusher.ForTestingPurposesOnly().AllowToFlushEvenIfManualFlushingSet.Add(Env);
+
+            using (Env.ReadTransaction()) // we're holding open read transaction so it won't be able to flush everything
+            {
+
+                ownFlusher.ForTestingPurposesOnly().AddEnvironmentToFlushQueue(Env);
+
+                ownFlusher.ForTestingPurposesOnly().ForceFlushEnvironment();
+
+                Assert.True(onLogsAppliedMre.Wait(Debugger.IsAttached ? TimeSpan.FromMinutes(5) : TimeSpan.FromSeconds(30)));
+            }
+
+            Assert.True(Env.Journal.Applicator.ShouldFlush);
+
+            onLogsAppliedMre.Reset();
+
+
+            ownFlusher.ForTestingPurposesOnly().AddEnvironmentToFlushQueue(Env);
+
+            // below two lines are necessary
+            Thread.Sleep(1000);
+            Env.TimeToSyncAfterFlushInSec = 0;
```

Comment:
Why is this?

Reply:
Sorry, forgot to finish the comments. We have internal checks which makes sure we don't flush too often. Anyway it should be set in the test during `Configure()` call

Reply:
No, we cannot set it via options because options set it to 30 seconds at minimum. Added the comment which explains why we have this.